### DISCUSSION TITLE
Ability to purge the cache of a specific route (or even a whole project)

### DISCUSSION THREAD

**Discussion by bokub** [vercel/community#1025] [Idea]
Hi,

On one of my projects, I have a serverless function (let's say `api/info/:id`) which:

- Is called very often by a lot of different visitors
- Should always return the same result for a given ID
- Can be quite long to respond

The response of the function for each ID will rarely change (like once a month). However, when something in my database is updated, every visitor should receive a fresh response quickly.

For now, I came up with the following strategy: the API responds with the following header: `Cache-Control: s-maxage=15, stale-while-revalidate=604800`. So every response comes from the cache, but this cache will be refreshed if older than 15 seconds.

**This works, but is not very efficient**, because the serverless function can be invocated up to 4 times / minutes for each ID in the worst-case scenario, and this is useless.

What would be great would be a way to force a cache purge for a given route (or even for the whole project)

I imagine it would work like this:

- A visitor calls `api/info/123`
- The route responds with `Cache-Control: s-maxage=86400` so Vercel caches the response for a whole day
- If ID 123 needs a fresh response, I'd manually purge the cache (for example I could call the route with some special header :shrug: )
- The next visitor who calls `api/info/123` will receive a fresh response of a response from a fresh cache

This is the only thing I could find about cache purging:

![image](https://user-images.githubusercontent.com/17952318/199711351-cb6f5deb-435c-40d6-bb29-c08f6f84df22.png)

Is there a way (even if it's some kind of hack) to force a cache purge for a single route **or even for the whole project**, without the need to redeploy my project?

Thank you very much :+1:

**Comment by amyegan** [comment]
Thanks for sharing this idea @bokub! I think [On-Demand Revalidation](https://vercel.com/docs/concepts/incremental-static-regeneration/quickstart#on-demand-revalidation) is the solution you're looking for. That will let you purge the cache using an API Route. 

There is also the option to bypass the cache by including a `?_vercel_no_cache=1` request parameter.

**Reply by bokub** [reply]
Thanks Amy !

I was aware of the cache bypass methods, but they **do not refresh the existing** cache, so they're useless for this specific problem.

However, I wasn't aware of the (quite new) Build Output API. I don't use Next.js, but a serverless function with the [nuxt builder](https://github.com/nuxt/vercel-builder), I'll try to see if I can switch to Build Output API!

I'll keep you updated! Thanks again :+1:
